src/sentry/spans/consumers/process/factory.py

```diff
@@ -145,14 +146,16 @@ def process_batch(
             with metrics.timer("spans.buffer.process_batch.decode"):
                 val = cast(SpanEvent, orjson.loads(payload.value))
 
-            if killswitches.killswitch_matches_context(
+            if killswitches.value_matches(
                 "spans.drop-in-buffer",
+                killswitch_config,
                 {
                     "org_id": val.get("organization_id"),
                     "project_id": val.get("project_id"),
                     "trace_id": val.get("trace_id"),
                     "partition_id": value.partition.index,
                 },
+                emit_metrics=False,
```

Comment:
If we expect that metrics are expensive, we also shouldn't log a timer per span just above. We can leave this for a follow up though, if you prefer.